diff -uprN a/arch/arm64/boot/dts/qcom/ipq8074.dtsi b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi	2024-01-27 20:59:39.052539000 -0800
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi	2024-01-27 21:29:02.553996627 -0800
@@ -560,13 +560,6 @@
 				bias-disable;
 			};
 
-			i2c_0_pins: i2c-0-state {
-				pins = "gpio42", "gpio43";
-				function = "blsp1_i2c";
-				drive-strength = <8>;
-				bias-disable;
-			};
-
 			spi_0_pins: spi-0-state {
 				pins = "gpio38", "gpio39", "gpio40", "gpio41";
 				function = "blsp0_spi";
@@ -574,13 +567,6 @@
 				bias-disable;
 			};
 
-			hsuart_pins: hsuart-state {
-				pins = "gpio46", "gpio47", "gpio48", "gpio49";
-				function = "blsp2_uart";
-				drive-strength = <8>;
-				bias-disable;
-			};
-
 			qpic_pins: qpic-state {
 				pins = "gpio1", "gpio3", "gpio4",
 				       "gpio5", "gpio6", "gpio7",
@@ -686,8 +672,6 @@
 			dmas = <&blsp_dma 4>,
 				<&blsp_dma 5>;
 			dma-names = "tx", "rx";
-			pinctrl-0 = <&hsuart_pins>;
-			pinctrl-names = "default";
 			status = "disabled";
 		};
 
@@ -732,8 +716,6 @@
 			clock-frequency = <400000>;
 			dmas = <&blsp_dma 14>, <&blsp_dma 15>;
 			dma-names = "tx", "rx";
-			pinctrl-0 = <&i2c_0_pins>;
-			pinctrl-names = "default";
 			status = "disabled";
 		};
 
