diff -uprN a/arch/arm64/boot/dts/qcom/ipq8074.dtsi b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi	2024-04-06 00:56:22.759003000 -0700
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi	2024-04-06 01:15:21.011122901 -0700
@@ -562,13 +562,6 @@
 				bias-disable;
 			};
 
-			i2c_0_pins: i2c-0-state {
-				pins = "gpio42", "gpio43";
-				function = "blsp1_i2c";
-				drive-strength = <8>;
-				bias-disable;
-			};
-
 			spi_0_pins: spi-0-state {
 				pins = "gpio38", "gpio39", "gpio40", "gpio41";
 				function = "blsp0_spi";
@@ -576,13 +569,6 @@
 				bias-disable;
 			};
 
-			hsuart_pins: hsuart-state {
-				pins = "gpio46", "gpio47", "gpio48", "gpio49";
-				function = "blsp2_uart";
-				drive-strength = <8>;
-				bias-disable;
-			};
-
 			qpic_pins: qpic-state {
 				pins = "gpio1", "gpio3", "gpio4",
 				       "gpio5", "gpio6", "gpio7",
@@ -692,8 +678,6 @@
 			dmas = <&blsp_dma 4>,
 				<&blsp_dma 5>;
 			dma-names = "tx", "rx";
-			pinctrl-0 = <&hsuart_pins>;
-			pinctrl-names = "default";
 			status = "disabled";
 		};
 
@@ -708,6 +692,16 @@
 			pinctrl-names = "default";
 			status = "disabled";
 		};
+		
+		blsp1_uart6: serial@78b4000 {
+			compatible = "qcom,msm-uartdm-v1.4", "qcom,msm-uartdm";
+			reg = <0x078b4000 0x200>;
+			interrupts = <GIC_SPI 309 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&gcc GCC_BLSP1_UART6_APPS_CLK>,
+				 <&gcc GCC_BLSP1_AHB_CLK>;
+			clock-names = "core", "iface";
+			status = "disabled";
+		};	
 
 		blsp1_spi1: spi@78b5000 {
 			compatible = "qcom,spi-qup-v2.2.1";
@@ -736,9 +730,6 @@
 			clock-names = "core", "iface";
 			clock-frequency = <400000>;
 			dmas = <&blsp_dma 14>, <&blsp_dma 15>;
-			dma-names = "tx", "rx";
-			pinctrl-0 = <&i2c_0_pins>;
-			pinctrl-names = "default";
 			status = "disabled";
 		};
 
